sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -171,6 +171,12 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
       val (storageFormat, provider) = getStorageFormatAndProvider(
         c.tableSpec.provider, c.tableSpec.options, c.tableSpec.location, c.tableSpec.serde,
         ctas = false)
+
+      if (GeneratedColumn.hasGeneratedColumns(c.tableSchema) &&
```

Comment:
Related to https://github.com/apache/spark/pull/38823/files#r1039264107 . We can easily get the `TableCatalog` here, as `CreateTable.name` is a `ResolvedIdentifier(catalog: TableCatalog, ...)`.

Reply:
Is there a reason why we match `ResolvedV1Identifier` here instead of `ResolvedV1TableIdentifier`?

I think we would want to add the API to `TableCatalog` and not `CatalogPlugin` right?